src/couch_mrview.erl

```diff
@@ -52,24 +52,31 @@
 %% In the most common case name is the view name and
 %% value is an object that must have a "map" function,
 %% and an optional "reduce" function. "map" and "reduce"
-%% values should be strings (function contents).
+%% values should be strings (function contents), except
+%% in case when langauge is <<query>> then maps must
+%% be objects.
 %%
 %% "lib" is a special case. The value
 %% of "lib" is an object that will contain libary code
 %% so it not validated.
 %%
-validate_view(<<"lib">>, {_Libs})->
+validate_view(<<"lib">>, {_Libs}, _)->
     ok;
-validate_view(VName, {Views}) ->
-    case couch_util:get_value(<<"map">>, Views) of
-        MapVal when is_binary(MapVal) ->
+validate_view(VName, {Views}, Language) ->
+    case {couch_util:get_value(<<"map">>, Views), Language}  of
+        {{[_ | _]}, <<"query">>} ->
             ok;
-        undefined ->
+        {Mapval, Lang} when is_binary(Mapval) andalso Lang =/= <<"query">> ->
+            ok;
+        {undefined, _} ->
             Err0 = <<"View ",VName/binary, " must have a map function">>,
             throw ({invalid_design_doc, Err0});
-        _ ->
-            Err1 = <<"`map` in ", VName/binary, " must be a string">>,
-            throw({invalid_design_doc, Err1})
+        {_, <<"query">>} ->
```

Comment:
aha, well, I think just pull this one up to be the second clause and we can drop the `andalso Lang =/= <<"query">>` bit?